vars/rapidastUtils.groovy

```diff
@@ -60,27 +60,40 @@ def prepareRapidastStages(String ServiceName, String ApiScanner, String TargetUr
 def parse_rapidast_options(String ServiceName, String ApiScanner, String TargetUrl, String ApISpecUrl) {
     // Parse the options for rapidast and add it to the config file. Always pull the latest config file
 
-    git url: 'https://github.com/RedHatProductSecurity/rapidast.git', branch: 'development'
+    git url: 'https://github.com/RedHatProductSecurity/rapidast.git', branch: 'main'
     def filename = 'config/config-template-long.yaml'
 
     // Comment the fields not required.
     sh "sed -i 's/importUrlsFromFile:/# importUrlsFromFile:/' ${filename}"
     sh "sed -i 's/defectDojoExport:/# defectDojoExport:/' ${filename}"
     sh "sed -i 's/# format:/format:/' ${filename}"
-    if ("${ApiScanner}" == "OpenApiScan") {
-        // Comment the fields not required.
+    sh "sed -i 's/spiderAjax:/# spiderAjax:/' ${filename}"
+    sh "sed -i 's/spider:/# spider:/' ${filename}"
+    sh "sed -i 's/maxDuration:/# maxDuration:/g' ${filename}"
+    sh "sed -i 's/browserId:/# browserId:/' ${filename}"
+    sh "sed -i 's/url:/# url:/g' ${filename}"
+    if ("${ApiScanner}" == "apiScan") {
         echo "OpenAPI Spec Compliant API Scan selected"
         sh "sed -i 's/graphql:/# graphql:/' ${filename}"
         sh "sed -i 's/spiderAjax:/# spiderAjax:/' ${filename}"
         sh "sed -i 's/spider:/# spider:/' ${filename}"
+        data = readYaml file: filename
+        data.scanners.zap.apiScan.target = "${TargetUrl}"
+        data.scanners.zap.apiScan.apis.apiUrl = "${OpenApISpecUrl}"
+    }
+    else if ("${ApiScanner} == graphql") {
+        sh "sed -i 's/apiScan:/# apiScan:/' ${filename}"
+        sh "sed -i 's/target:/# target:/' ${filename}"
+        sh "sed -i 's/apis:/# apis:/' ${filename}"
+        sh "sed -i 's/apiUrl:/# apiUrl:/' ${filename}"
+        sh "sed -i 's/# schemaUrl:/schemaUrl:/' ${filename}"
+        data = readYaml file: filename
+        data.scanners.zap.graphql.endpoint = "${TargetUrl}"
+        data.scanners.zap.graphql.schemaUrl = "${OpenApISpecUrl}"
     }
     else {
-        echo "Scanner not supported."
-        currentBuild.result = 'FAILURE'
-        sh "exit 1"
+        echo "Scanner not supported"
     }
```

Comment:
It looks like all these lines could be refactored into a single script (that can run outside of Jenkins) , taking the "ApiScanner" as a parameter.

Reply:
This is run inside the pod, and wanted to keep logic in single place if , someone wants to setup their own job. For context this is part of https://github.com/RedHatInsights/iqe-jenkins/blob/master/jenkins_job_dsl/src/jobs/internal/rapidastJobDeployer.groovy

Reply:
We can keep it here for now